zebra-script/src/lib.rs

```diff
@@ -226,6 +233,49 @@ impl CachedFfiTransaction {
             Err(Error::from(err))
         }
     }
+
+    /// Returns the destination address of the transparent output at `output_index`.
+    pub fn transparent_output_address(
+        &self,
+        network: Network,
+        output_index: usize,
+    ) -> Result<transparent::Address, Error> {
+        let mut err = 0;
+        let mut addr_type = 0;
+
+        // This conversion is useful on some platforms, but not others.
+        #[allow(clippy::useless_conversion)]
+        let n_out = output_index
+            .try_into()
+            .expect("transaction indexes are much less than c_uint::MAX");
+
+        let address_hash = unsafe {
```

Comment:
Safety: Can we write a safety comment here?
```suggestion
        // SAFETY:
        // `CachedFfiTransaction::new` makes sure `self.precomputed` is not NULL.
        // `n_out` is checked by the called function.
        // All other arguments are local stack references, which are always valid.
        let address_hash = unsafe {
```